Source/WebCore/platform/mediastream/RealtimeVideoCaptureSource.cpp

```diff
@@ -489,27 +489,30 @@ auto RealtimeVideoCaptureSource::takePhoto(PhotoSettings&& photoSettings) -> Ref
 
         m_currentPreset = newPresetForPhoto->encodingPreset;
         auto newSize = newPresetForPhoto->encodingPreset->size();
+        startApplyingConstraints();
```

Comment:
We could cover these changes by overriding in MockRealtimeVideoSource startApplyingConstraints/endApplyingConstraints/setFrameRateAndZoomWithPreset to set a boolean in the first two and assert on the boolean in the third one.